usr/src/cmd/altexec/altexec.c

```diff
@@ -117,7 +109,8 @@ generate_pkgsrc_path(const char *desired_bin)
 	 */
 
 	if (asprintf(&binpath, "%s/%s", path, desired_bin) == -1) {
```

Comment:
So this only fails if we can't allocate enough memory for the buffer, which should be (by my calculation) up to 21b. And if we can't allocate a measly 21b, then the whole system is in a pretty dire state?

Reply:
Agreed.  So I'll lose the printf (but keep the assert()).